lib/device_detector/version_extractor.rb

```diff
@@ -18,13 +18,17 @@ def extract_version
     end
 
     def replace_version_string_with(match_data)
-      string = version_string
+      string = version_string.to_s
 
       1.upto(9) do |index|
-        string = string.gsub(/\$#{index}/, match_data[index]) if match_data[index]
+        if match_data[index]
+          string = string.gsub(/\$#{index}/, match_data[index])
+        else
+          string = string.gsub(/\$#{index}/, '')
+        end
       end
 
-      string
+      string.strip
```

Comment:
Could you explain how / when could it happen, that the string has some trailining whitespace in it?